manifests/stage-nv-peer-mem-driver/0050_nv-peer-mem-driver-ds.yaml

```diff
@@ -146,3 +146,10 @@ spec:
                 operator: Gt
                 values:
                   - "0"
+              # cuda driver starting from v465 should include builtin nvidia_peermem module
+              # which should be used instead nv_peer_mem
+              # prevent scheduling nv-peer-mem-driver POD on nodes with cuda driver version >= 465
+              - key: nvidia.com/cuda.driver.major
```

Comment:
A layman question here: when cuda driver get updated and this condition is updated to be no longer true, will the pod get unscheduled from the node ?

the term `requiredDuringSchedulingIgnoredDuringExecution` has me a bit worried.

Reply:
> will the pod get unscheduled from the node ?

Yes

Reply:
hmm so i wonder why we have `IgnoredDuringExecution`.

according to k8s documentation it shouldnt evict the pod

>Node affinity is conceptually similar to nodeSelector -- it allows you to constrain which nodes your pod is eligible to be scheduled on, based on labels on the node.
>
>There are currently two types of node affinity, called requiredDuringSchedulingIgnoredDuringExecution and preferredDuringSchedulingIgnoredDuringExecution. You can think of them as "hard" and "soft" respectively, in the sense that the former specifies rules that must be met for a pod to be scheduled onto a node (similar to nodeSelector but using a more expressive syntax), while the latter specifies preferences that the scheduler will try to enforce but will not guarantee.

And here is the more relevant part:

> The "IgnoredDuringExecution" part of the names means that, similar to how nodeSelector works, if labels on a node change at runtime such that the affinity rules on a pod are no longer met, the pod continues to run on the node. In the future we plan to offer requiredDuringSchedulingRequiredDuringExecution which will be identical to requiredDuringSchedulingIgnoredDuringExecution except that it will evict pods from nodes that cease to satisfy the pods' node affinity requirements.

Reply:
nv-peer-mem-driver is a DaemonSet.
DaemonSet controller manage PODs creation and removal 
Check this doc section: https://kubernetes.io/docs/concepts/workloads/controllers/daemonset/#updating-a-daemonset

Reply:
nice ! thanks.